scripts/identify_paired_differences.py

```diff
@@ -159,6 +159,10 @@ def main():
 
     if biom_table_fp:
         biom_table = load_table(biom_table_fp)
+        sample_ids_to_keep = biom_table.ids(axis='sample')
+        for sid in mapping_data.keys():
+            if sid not in sample_ids_to_keep:
```

Comment:
This lookup is O(n) because `sample_ids_to_keep` is an `ndarray`. Does BIOM provide a quick way to check for IDs? If not you could create a `set`.